### ISSUE TITLE
Landmarks detection

### ISSUE THREAD

**Issue by ilyanelken** [AITTSMD/MTCNN-Tensorflow#15]
Hi @AITTSMD,

As you stated there was an issue with using CelebA dataset for training, due to some corrupted labels.
I see now that there is an issue with only "101283.jpg" image that has the following bounding box annotation: "101283.jpg   320 828   0   0" which is invalid.

All the rest of the records seems valid.

Regards,
Ilya

**Comment by AITTSMD** [comment]
@ilyanelken hi,are u sure that all the other records are valid?

**Comment by ilyanelken** [comment]
@AITTSMD yes

**Comment by ilyanelken** [comment]
@AITTSMD yes

**Comment by ilyanelken** [comment]
@AITTSMD I have no way to check that all records are valid, but if the invalid record I mentioned is removed, the training process might be executed, but the landmarks results are still not satisfying enough :(

**Comment by ilyanelken** [comment]
@AITTSMD also in:
https://github.com/AITTSMD/MTCNN-Tensorflow/blob/master/prepare_data/gen_hard_example.py#L195
the model epochs that are used by default are: [18, 14, 22], while the default end epochs for training are: [30, 22, 22]
Any special reason for that?

**Comment by AITTSMD** [comment]
@ilyanelken No special reason. What about the result of training landmark using CelebA.

**Comment by ilyanelken** [comment]
@AITTSMD Landmarks are still not satisfying. Also in:
https://github.com/AITTSMD/MTCNN-Tensorflow/blob/master/prepare_data/gen_imglist_pnet.py#L40
And the rest of similar lines, I suppose "replace" should be false.

**Comment by ilyanelken** [comment]
@AITTSMD I have few more fixes for proper landmarks training, are you going to accept pull requests?

**Comment by AITTSMD** [comment]
@ilyanelken Hi，where did you improve？I'm willing to accept if result gets better.

**Comment by ilyanelken** [comment]
@AITTSMD Sounds reasonable. I will open a PR soon, you will be able to review it.
